### PR TITLE
[ADD] Added new module stock_kardex_report task#3957

### PR DESCRIPTION
**Pull request by Tino480**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (alan196) [inline .vscode/settings.json]: Please remove this file

File: stock_kardex_report/.vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (alan196) [inline stock_kardex_report/.vscode/settings.json]: please remove this file

File: stock_kardex_report/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2020, Jarsa Sistemas, S.A. de C.V.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lpgl.html).
+
+{
+    'name': 'Stock Kardex Report',
+    'summary': 'Generate Kardex Report',
+    'version': '13.0.',
```
Comment (alan196) [inline stock_kardex_report/__manifest__.py]: 13.0.1.0.0

**Review by alan196 — CHANGES_REQUESTED** [review]

**Comment by alan196** [comment]
@Tino480 Please copy the description of https://github.com/OCA/stock-logistics-reporting/pull/93#issuecomment-608154607 and also rename the title of this PR to define the version

File: stock_kardex_report/.vscode/settings.json (hunk 2 above)
Comment (Tino480) [reply stock_kardex_report/.vscode/settings.json]: Removed

File: .vscode/settings.json (hunk 1 above)
Comment (Tino480) [reply .vscode/settings.json]: Removed

**Comment by rousseldenis** [comment]
@Tino480 Could you check pre-commit : https://github.com/OCA/maintainer-tools/#integration-with-pre-commit

File: stock_kardex_report/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2020, Jarsa Sistemas, S.A. de C.V.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lpgl.html).
+
+{
+    "name": "Stock Kardex Report",
+    "summary": "Generate Kardex Report",
+    "version": "13.0.1.0.0",
+    "category": "Reports",
+    "author": "Jarsa Sistemas, S.A. de C.V.," "Odoo Community Association (OCA)",
+    "website": "https://www.jarsa.com.mx",
```
Comment (rousseldenis) [inline stock_kardex_report/__manifest__.py]: Please change this according to https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10

File: stock_kardex_report/models/stock_kardex_report.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    move_id = fields.Many2one("stock.move", readonly=True)
+    product_id = fields.Many2one("product.product", readonly=True)
+    product_uom_id = fields.Many2one("uom.uom", readonly=True)
+    lot_id = fields.Many2one("stock.production.lot", readonly=True)
+    owner_id = fields.Many2one("res.partner", readonly=True)
+    package_id = fields.Many2one("stock.quant.package", readonly=True)
+    location_id = fields.Many2one("stock.location", readonly=True)
+    location_dest_id = fields.Many2one("stock.location", readonly=True)
+    qty_done = fields.Float("Done", readonly=True)
```
Comment (rousseldenis) [inline stock_kardex_report/models/stock_kardex_report.py]: IMHO quantities (+ balance) should implement decimal precision.

File: stock_kardex_report/wizard/stock_kardex_report_wizard.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+class StockKardexReportWiz(models.TransientModel):
+    _name = "stock.kardex.report.wiz"
+    _description = "Wizard to create kardex reports of stock moves"
+
+    date_from = fields.Datetime(
+        string="From", required=True, default=fields.Datetime.now
+    )
+    date_to = fields.Datetime(string="To", required=True, default=fields.Datetime.now)
+    product = fields.Many2one("product.product", required=True)
+    location = fields.Many2one("stock.location", required=True)
```
Comment (rousseldenis) [inline stock_kardex_report/wizard/stock_kardex_report_wizard.py]: Please add ondelete="cascade"

File: .eslintrc [hunk 7]
```diff
@@ -5,7 +5,7 @@
     "fuzzy": false,
     "jQuery": false,
     "moment": false,
-    "odoo": false,
+    "odoo": true,
```
Comment (rousseldenis) [inline .eslintrc]: Why changing this ?

File: stock_kardex_report/__manifest__.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    "license": "LGPL-3",
+    "depends": ["stock"],
+    "data": [
+        "views/stock_kardex_report_views.xml",
+        "wizard/stock_kardex_report_wizard_view.xml",
+        "security/ir.model.access.csv",
```
Comment (rousseldenis) [inline stock_kardex_report/__manifest__.py]: I'd prefer xml declaration for security as it allows inheritance.

File: stock_kardex_report/models/stock_kardex_report.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+
+class StockKardexReport(models.Model):
+    _name = "stock.kardex.report"
+    _description = "This model creates a kardex report for stock moves"
+    _order = "date desc"
+
+    move_id = fields.Many2one("stock.move", readonly=True)
+    product_id = fields.Many2one("product.product", readonly=True)
+    product_uom_id = fields.Many2one("uom.uom", readonly=True)
```
Comment (rousseldenis) [inline stock_kardex_report/models/stock_kardex_report.py]: Please add meaningful strings for report column names:

image

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code and funcitonal review. Some comments.

File: stock_kardex_report/__manifest__.py (hunk 4 above)
Comment (Tino480) [reply stock_kardex_report/__manifest__.py]: Changed in new commit

File: stock_kardex_report/models/stock_kardex_report.py (hunk 5 above)
Comment (Tino480) [reply stock_kardex_report/models/stock_kardex_report.py]: Implemented decimal precision in new commit

File: stock_kardex_report/wizard/stock_kardex_report_wizard.py (hunk 6 above)
Comment (Tino480) [reply stock_kardex_report/wizard/stock_kardex_report_wizard.py]: Added ondelete="cascade" in new commit

File: .eslintrc (hunk 7 above)
Comment (Tino480) [reply .eslintrc]: I was having problems running pre-commit and a search recommended I change that I forgot to change it back... Changed it back with new commit

File: stock_kardex_report/__manifest__.py (hunk 8 above)
Comment (Tino480) [reply stock_kardex_report/__manifest__.py]: Declared it in xml in new commit

File: stock_kardex_report/models/stock_kardex_report.py (hunk 9 above)
Comment (Tino480) [reply stock_kardex_report/models/stock_kardex_report.py]: Changed column names in new commit

File: stock_kardex_report/security/stock_kardex_report_security.xml [hunk 10]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
```
Comment (alan196) [inline stock_kardex_report/security/stock_kardex_report_security.xml]: use `<odoo noupdate="1">`

**Review by alan196 — CHANGES_REQUESTED** [review]

**Comment by rousseldenis** [comment]
@Tino480 @alan196 Could you solve pre-commit issue running 'pre-commit run -a' locally before comitting ?

**Comment by alan196** [comment]
@rousseldenis tests are now green 👍

**Pull request closed without merging**